### PR TITLE
refactor(docs): clean up components.js

### PR DESCRIPTION
**Pull request by macklinu**
This is to help the discussion in https://github.com/ChristopherBiscardi/gatsby-mdx/issues/66#issuecomment-417679253. /cc @greggb

Talking about splitting the `components.js` file into a `src/components/` module directory, if we wanted to roll our own component directory, it could look something like this. This just refactors what we currently have to fit this sort of structure.

- `src/components/index.js` re-exports all internal components so you can just do something like `import { Pre, Code } from './src/components'`
- `src/components/theme.js` defines the `styled-system` theme
- `src/components/mdxProviderComponents.js` maps our design system to an object usable by the `<MDXProvider>` component

If we end up deciding to use something like Rebass instead, we can swap out individual component file implementations. So like:

```js
// before
// src/components/pre.js

import system from 'system-components/emotion'
const Pre = system(
  {
    is: 'pre',
    fontSize: 1,
    fontFamily: 'mono',
    m: 0,
  },
  {
    overflow: 'auto',
  },
  'fontFamily',
  'space',
  'color'
)
Pre.displayName = 'Pre'
export default Pre
```

```js
// after, using Rebass
// src/components/pre.js

import { Pre } from 'rebass'

const CustomPre = styled(Pre)`
  /* custom styles */
`

export default CustomPre
```

And the rest of our docs site could remain untouched.

Feedback welcome on this code structure or another one that will help flesh out the docs site. @greggb @ChristopherBiscardi and others

Also, proof that this refactor didn't break anything. 😄 

image

image

### CODE REVIEW

**Comment by ChristopherBiscardi** [comment]
Also for future reference, the GitHub checks have a link to a deploy preview for the docs site (and a few others). Clicking the `netlify/gatsby-mdx/deploy-preview` should take anyone to a deploy preview of the docs site for the PR

File: examples/docs/src/components/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export { default as theme } from "./theme";
+export { default as mdxProviderComponents } from "./mdxProviderComponents";
+export { default as ThemeProvider } from "./themeProvider";
+export { default as Code } from "./code";
+export { default as CodeBlock } from "./codeBlock";
+export { default as Container } from "./container";
+export { default as Heading } from "./heading";
+export { default as Notification } from "./notification";
+export { default as Pre } from "./pre";
+export { default as Text } from "./text";
```
Comment (greggb) [inline examples/docs/src/components/index.js]: Is this different than `export Text from './text'`? Maybe this avoids having to import first?

**Comment by greggb** [comment]
This is awesome @macklinu. The other pattern I've used is doing all exports from an index file within each components directory, but that is more relevant when there is a test file, sub components, and maybe a storybook file colocated. I like the simplicity of the imports with what you have here.

>If we end up deciding to use something like Rebass instead, we can swap out individual component file implementations

🎉 

I'm planning on two QOL PRs this weekend - one with Prettier* formatting support and one with an alias for the components directory `$components/Notification<or whatever>`. You all ok with those, @ChristopherBiscardi, @macklinu?

* 4 space indentation, es5 trailing comma, and single quotes is my usual config.

File: examples/docs/src/components/index.js (hunk 1 above)
Comment (macklinu) [reply examples/docs/src/components/index.js]: That's a good question, lemme try that out!

Comment (macklinu) [reply examples/docs/src/components/index.js]: Looks like we need to enable this syntax with a Babel plugin. Lemme create a follow-up issue / PR.

image

**Comment by macklinu** [comment]
@greggb I split out #103 and #104 as follow up issues to this PR. 

> 4 space indentation, es5 trailing comma, and single quotes is my usual config

We already have Prettier support in the codebase, which uses the default Prettier config. Since I'm not the owner of the repo, I don't want to call the shots in terms of how code is formatted (I'm more of a 2-space, es5 trailing comma, single quotes, no semicolon guy myself), but perhaps we can discuss a change to the configuration in a follow up issue or PR with @ChristopherBiscardi.

**Comment by ChristopherBiscardi** [comment]
I prefer the default prettier config because it rules out potential tooling problems if someone's $editor/CLI script/etc doesn't pick up config in a specific place. Which makes it easier for beginners who may not understand how to configure things yet.

Prettier should already be applied via git commit hook (via husky/lint-staged) although we don't have anything checking for that in CI.

To me one of the benefits of prettier is that anyone can set up their editor with any prettier config they like (4 spaces, 2 spaces, quotes, commas, etc) while working on a PR, and the core repo automatically gets a single style in PRs, etc. In this way, I'm not sure the core style matters as long as it's consistent. 

In the end, I personally I don't care what the prettier config is as long as it's not something I have to think about and is automatically applied :)

**Comment by greggb** [comment]
No worries, I think I have my viscose to only `format-on-save` if there is a prettier config somewhere in the repo. I'll just disable that for this workspace so it'll use the defaults.

**Pull request merged**